src/Entity/ItemId.php

```diff
@@ -95,16 +96,18 @@ public function unserialize( $serialized ) {
 	 * serialization. Not doing so in new code requires special justification.
 	 *
 	 * @param int|float|string $numericId
+	 * @param string $repositoryName, defaults to an empty string (local repository)
 	 *
 	 * @return self
 	 * @throws InvalidArgumentException
 	 */
-	public static function newFromNumber( $numericId ) {
+	public static function newFromNumber( $numericId, $repositoryName = '' ) {
 		if ( !is_numeric( $numericId ) ) {
 			throw new InvalidArgumentException( '$numericId must be numeric' );
 		}
+		RepositoryNameAssert::assertParameterIsValidRepositoryName( $repositoryName, '$repositoryName' );
```

Comment:
This throws `ParameterAssertionException`, not `InvalidArgumentException`, so the constructor doc is now wrong.

As a user of a method such as this one, I kinda expect to get a `InvalidArgumentException` when the argument is invalid. Forcing the user to bind to the assertion library used is not nice.